Source/Diagnostics/ReducedDiags/RawEFieldReduction.H

```diff
@@ -0,0 +1,275 @@
+/* Copyright 2021 Revathi Jambunathan
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_DIAGNOSTICS_REDUCEDDIAGS_RAWEFIELDREDUCTION_H_
+#define WARPX_DIAGNOSTICS_REDUCEDDIAGS_RAWEFIELDREDUCTION_H_
+
+#include "ReducedDiags.H"
+#include "WarpX.H"
+
+#include <AMReX_Array.H>
+#include <AMReX_Box.H>
+#include <AMReX_Config.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_FabArray.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_GpuControl.H>
+#include <AMReX_GpuQualifiers.H>
+#include <AMReX_IndexType.H>
+#include <AMReX_MFIter.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_Parser.H>
+#include <AMReX_REAL.H>
+#include <AMReX_RealBox.H>
+#include <AMReX_Reduce.H>
+#include <AMReX_Tuple.H>
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+
+class RawEFieldReduction : public ReducedDiags
+{
+public:
+
+    /**
+     * constructor
+     * @param[in] rd_name reduced diags names
+     */
+    RawEFieldReduction(std::string rd_name);
+
+    /**
+     * This function is called at every time step, and if necessary calls the templated function
+     * ComputeRawEFieldReduction(), which does the actual reduction computation.
+     *
+     * @param[in] step the timestep
+     */
+    virtual void ComputeDiags(int step) override final;
+
+private:
+    /// Parser to read expression to be reduced from the input file.
+    /// 3 elements are x, y, z
+    static constexpr int m_nvars = 3;
+    std::unique_ptr<amrex::Parser> m_parser;
+
+    // Type of reduction (e.g. Maximum, Minimum or Sum)
+    int m_reduction_type;
+    // Type of integration (e.g. volume or surface)
+    int m_integral_type;
+#if (AMREX_SPACEDIM==2)
+    int m_surface_normal[2]={0,0};
```

Comment:
Please can you add comments to explain the new integer member variable added

Reply:
I did.